include/itkAttenuationImageFilter.hxx

```diff
@@ -0,0 +1,225 @@
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::AttenuationImageFilter()
+{
+  this->SetNumberOfRequiredInputs(1);
+  this->DynamicMultiThreadingOn();
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+const ImageRegionSplitterBase *
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::GetImageRegionSplitter() const
+{
+  m_RegionSplitter->SetDirection(m_Direction);
+  return m_RegionSplitter.GetPointer();
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::BeforeThreadedGenerateData()
+{
+  // Verify inputs
+  if (this->GetMaskImage() == nullptr)
+  {
+    itkExceptionMacro("Filter requires a mask image for inclusion estimates!");
+  }
+
+  if (this->GetDirection() > ImageDimension)
+  {
+    itkExceptionMacro("Scan line direction must be a valid image dimension!");
+  }
+
+  if (this->GetSamplingFrequencyMHz() < itk::Math::eps)
+  {
+    itkExceptionMacro("RF sampling frequency was not set!");
+  }
+
+  Superclass::BeforeThreadedGenerateData();
+
+  // Initialize metric image
+  this->GetOutput()->Allocate();
+  this->GetOutput()->FillBuffer(0.0f);
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::AfterThreadedGenerateData()
+{
+  Superclass::AfterThreadedGenerateData();
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::DynamicThreadedGenerateData(
+  const OutputRegionType & regionForThread)
+{
+  if (regionForThread.GetNumberOfPixels() == 0)
+  {
+    return;
+  }
+
+  const MaskImageType *  maskImage = this->GetMaskImage();
+  const InputImageType * input = this->GetInput();
+  OutputImageType *      output = this->GetOutput();
+
+  ImageLinearConstIteratorWithIndex<TInputImage> it(input, regionForThread);
+  it.SetDirection(m_Direction);
+  it.GoToBegin();
+
+  unsigned int   inclusionLength;
+  InputIndexType start, end;
+
+  const float scanStepMM = input->GetSpacing()[m_Direction];
+
+  // do the work
+  while (!it.IsAtEnd())
+  {
+    inclusionLength = 0;
+    while (!it.IsAtEndOfLine())
+    {
+      // Advance until an inclusion is found
+      InputIndexType index = it.GetIndex();
+      if (maskImage->GetPixel(index) == m_LabelValue && inclusionLength == 0)
+      {
+        // Step into inclusion
+        start = it.GetIndex();
+      }
+      else if (maskImage->GetPixel(index) != m_LabelValue && inclusionLength > 0)
+      {
+        // Step out of inclusion
+        end = it.GetIndex();
+
+        // Adjust for pixel padding
+        start[m_Direction] += m_PadPixelUpperBounds;
+        end[m_Direction] -= m_PadPixelLowerBounds;
+
+        // Adjust for spatial padding
+        start[m_Direction] += static_cast<unsigned int>(m_PadSpatialUpperBoundsMM / scanStepMM);
```

Comment:
These two lines should be removed.

Reply:
Do we want to keep pixel-based padding to slightly ignore/erode the edges of the inclusion? If so, it's relatively easy to add an additional interface method to set that pixel width via a spatial distance, provided that an input image with associated spacing is provided. These lines could be removed while essentially retaining their function in pixel edge padding.

Reply:
Yes, we want to keep pixel-based padding. We could also provide a convenience accessors in millimeters.